diff --git a/src/fr-window.c b/src/fr-window.c
index 0c06cb65..6f6b4b99 100644
--- a/src/fr-window.c
+++ b/src/fr-window.c
@@ -9172,8 +9172,9 @@ monitor_extracted_files (OpenFilesData *odata)
 }
 
 
+#ifdef USE_NONNATIVE_APPCHOOSER
 static gboolean
-fr_window_open_extracted_files (OpenFilesData *odata)
+open_extracted_files_with_nonnative_appchooser (OpenFilesData *odata)
 {
 	GList               *file_list = odata->cdata->file_list;
 	GFile               *first_file;
@@ -9251,6 +9252,53 @@ fr_window_open_extracted_files (OpenFilesData *odata)
 
 	return result;
 }
+#else
+static void
+open_extracted_files_with_native_appchooser (OpenFilesData *odata)
+{
+	GList               *file_list = odata->cdata->file_list;
+	GList               *scan;
+	GError              *error = NULL;
+	GtkWindow           *window;
+	char                *uri;
+
+	g_return_if_fail (file_list != NULL);
+
+	window = GTK_WINDOW (odata->window);
+
+	if (! odata->window->archive->read_only)
+		monitor_extracted_files (odata);
+
+	if (odata->ask_application) {
+		dlg_open_with (odata->window, file_list);
+		return;
+	}
+
+	for (scan = file_list; scan; scan = scan->next) {
+		uri = g_file_get_uri (G_FILE (scan->data));
+		if (!gtk_show_uri_on_window (window, uri, GDK_CURRENT_TIME, &error)) {
+			_gtk_error_dialog_run (window,
+					_("Could not perform the operation"),
+					"%s",
+					error->message);
+			g_clear_error (&error);
+		}
+	}
+
+	g_free (uri);
+}
+#endif
+
+
+static void
+fr_window_open_extracted_files (OpenFilesData *odata)
+{
+#ifdef USE_NONNATIVE_APPCHOOSER
+	open_extracted_files_with_nonnative_appchooser (odata);
+#else
+	open_extracted_files_with_native_appchooser (odata);
+#endif
+}
 
 
 static void
